pages/admin/dashboard/builder.tsx

```diff
@@ -0,0 +1,268 @@
+/* eslint-disable prettier/prettier */
+import React, { useState } from "react";
+import {
+    Flex,
+    Text,
+    Button,
+    Spacer,
+    Stack,
+    Box,
+    useDisclosure,
+    VStack,
+    Heading,
+    HStack,
+    Container,
+    IconButton,
+    ButtonGroup,
+    Editable,
+    EditablePreview,
+    EditableInput,
+} from "@chakra-ui/react";
+import { IoIosUndo, IoIosRedo } from "react-icons/io";
+import { IoTrash, IoDesktopOutline } from "react-icons/io5";
+import { FaMobileAlt } from "react-icons/fa";
+
+import { Page } from "types/Page";
+import {
+    BuilderLayout,
+    MarkdownEditor,
+    MarkdownRenderer,
+    ModulePreview,
+    CheckboxComp,
+} from "@components";
+
+const Builder: Page = () => {
+    const {
+        isOpen: isSidebarOpen,
+        onToggle: toggleSidebar,
+        onOpen: sidebarOpen,
+        onClose: sidebarClose,
+    } = useDisclosure();
+    const [editorText, setEditorText] = useState("Hello world!");
+    const [slideNumber, setSlide] = useState(1);
+    const [maxSlides, addSlide] = useState(1);
+    const [prevSlide, setPrevSlide] = useState(1);
+    const [slides, setSlides] = useState([editorText]);
+    const [prevButton, setPrevButton] = useState(true);
+    const [nextButton, setNextButton] = useState(true);
+    const [printButton, setPrintButton] = useState(false);
+    const [saveButton, setSaveButton] = useState(false);
```

Comment:
Perhaps this isn't the most scalable way to do this - maybe we can have just one state variable which is a Set which stores strings?

something like
`const [buttons, setButtons] = useState(new Set(['prev', 'next']))`

We'd also want to have a separate constant object acting as an enum which stores the strings corresponding to each string

Reply:
That object could also contain the labelling text info as well - two birds in one stone :D

Reply:
Thanks for the review, Chamod. I made the changes with the buttons in a set like you suggested :)